src/dbnode/persist/fs/retriever.go

```diff
@@ -168,6 +169,12 @@ func (r *blockRetriever) CacheShardIndices(shards []uint32) error {
 	return seekerMgr.CacheShardIndices(shards)
 }
 
+func (r *blockRetriever) AssignShardSet(shardSet sharding.ShardSet) {
+	if seekerMgr := r.seekerMgr; seekerMgr != nil {
+		seekerMgr.AssignShardSet(shardSet)
+	}
```

Comment:
Hm, think we should probably cache it locally if it's nil and apply it after creating the seeker manager. 

We should also require it on the `Open` methods of both `blockRetriever` and `seekerManager`, so that there's always a shard set set from the moment they're open (and can reliably depend on having it available in the seeker manager.

i.e. make it consistent.
```go
r.Lock()
defer r.Unlock()

if r.seekerMgr != nil {
  r.seekerMgr.AssignShardSet(shardSet)
}

```

Reply:
So I verified that the `r.seekerMgr` will always be non `nil` before it's possible to call `AssignShardSet`. 

As we always open a block retriever when creating one for a ns.
```
blockRetrieverMgr := block.NewDatabaseBlockRetrieverManager(
        func(md namespace.Metadata) (block.DatabaseBlockRetriever, error) {
                retriever, err := fs.NewBlockRetriever(retrieverOpts, fsopts)
                if err != nil {
                        return nil, err
                }
                if err := retriever.Open(md); err != nil {
                        return nil, err
                }
                return retriever, nil
        })
```

Reply:
I was initializing shardSet to empty in the seek manager but I can pass it in through `Open()`. I don't think the block retriever should store a shardSet in it's state though as it doesn't use it.

Reply:
Agreed on "don't think the block retriever should store a shardSet in it's state though as it doesn't use it", do think it's worth passing it through Open though perhaps otherwise the seeker manager is not able to be used in any meaningful way.